noxfile.py

```diff
@@ -5,28 +5,51 @@
 
 import nox
 
-nox.options.reuse_existing_virtualenvs = True
+nox.options.reuse_existing_virtualenvs = False
 
+# Sphinx output and source directories
+BUILD_DIR = "_build"
+OUTPUT_DIR = pathlib.Path(BUILD_DIR, "html")
+SOURCE_DIR = pathlib.Path("docs")
 
-# Use this for venv envs nox -s test
-@nox.session(python=["3.10", "3.11", "3.12"])
-def tests(session):
-    """Install requirements in a venv and run tests."""
-    session.install(".[tests]")
-    session.run(
-        "pytest",
-        "--cov=src/stravalib",
-        "--cov-report=xml:coverage.xml",
-        "--cov-report=term",
-        "src/stravalib/tests/unit/",
-        "src/stravalib/tests/integration/",
-    )
+# Sphinx build commands
+SPHINX_BUILD = "sphinx-build"
+SPHINX_AUTO_BUILD = "sphinx-autobuild"
+
+# Sphinx parameters used to build the guide
+BUILD_PARAMETERS = ["-b", "html"]
+
+# Sphinx parameters used to test the build of the guide
+TEST_PARAMETERS = ["-W", "--keep-going", "-E", "-a"]
 
+# Sphinx-autobuild ignore and include parameters
+AUTOBUILD_IGNORE = [
+    "_build",
+    ".nox",
+    "build_assets",
+    "tmp",
+]
+AUTOBUILD_INCLUDE = [pathlib.Path("_static", "pyos.css")]
 
 # Build docs
 build_command = ["-b", "html", "docs/", "docs/_build/html"]
 
 
+@nox.session(name="docs-test", python="3.11")
+def docs_test(session):
```

Comment:
I added a small test method here to allow us to more easily troubleshoot locally. there is still the sphinx duplicate index warning that I can't yet figure out how to turn off! but this is useful locally to identify issues